plugins/capture.rb

```diff
@@ -0,0 +1,612 @@
+        if self.respond_to?(method)
+          self.send(method, datastore, config)
+        end
+
+        if encrypted.include?(svc)
+          configure_tls(datastore, config)
+        end
+
+        # Before running everything, let's do some basic validation of settings
+        mod_dup = mod.replicant
+        mod_dup._import_extra_options(opts)
+        mod_dup.options.validate(mod_dup.datastore)
+
+        modules_to_run.append([svc, mod, opts])
+      end
+
+      modules_to_run.each do |svc, mod, opts|
+        event = Rex::Sync::Event.new(state=false,auto_reset=false)
+        job_listener = CaptureJobListener.new(mod.name, event)
+
+
+        result = Msf::Simple::Auxiliary.run_simple(mod, opts, job_listener: job_listener)
+        job_id = result[1]
+
+        # Wait for the event to trigger (socket server either waiting, or failed)
+        event.wait
+        if job_listener.succeeded
+          # Keep track of it so we can close it upon a `stop` command
+          @active_job_ids[session].append(job_id)
+          job = framework.jobs[job_id.to_s]
+          # Rename the job for display (to differentiate between the encrypted/plaintext ones in particular)
+          if config[:session].nil?
+            session_str = 'local'
+          else
+            session_str = "session #{config[:session].to_i}"
+          end
+          job.send(:name=, "Capture (#{session_str}): #{svc}")
+        end
+      end
+
+      print_good('Started capture jobs')
+    end
+
+    def listeners_stop(args)
+      options = parse_stop_args(args)
+      if options[:show_help]
+        help('stop')
+        return
+      end
+
+      session = options[:session]
+      job_id_clone = @active_job_ids.clone
+      job_id_clone.each do |session_id, jobs|
+        if session.nil? || session == session_id
+          jobs.each do | job_id|
+            framework.jobs.stop_job(job_id) unless framework.jobs[job_id.to_s].nil?
+          end
+          jobs.clear
+          @active_job_ids.delete(session_id)
+        end
+      end
+
+      loggers_clone = @active_loggers.clone
+      loggers_clone.each do |session_id, logger|
+        if session.nil? || session == session_id
+          logger.close
+          @active_loggers.delete(session_id)
+        end
+      end
+
+      print_line('Capture listeners stopped')
+    end
+
+    # Print the appropriate help text depending on an optional option parser.
+    #
+    # @param first_arg [String] the first argument to this command
+    # @return [nil]
+    def help(first_arg = nil)
+      if first_arg == 'start'
+        print_line('Usage: capture start -i <ip> [options]')
+        print_line(@@start_opt_parser.usage)
+      elsif first_arg == 'stop'
+        print_line('Usage: capture stop [options]')
+        print_line(@@stop_opt_parser.usage)
+      else
+        print_line('Usage: capture [start|stop] [options]')
+      end
+    end
+
+    def default_options
+        options = {
+         :ntlm_challenge => nil,
```

Comment:
Running rubocop with `rubocop -A plugins/capture.rb` should be able to replace these with `ntlm_challenge: nil`, as well as a few other linting things in the file :+1: